flight_backend/api/serializers/flight.py

```diff
@@ -0,0 +1,96 @@
+from rest_framework import serializers
+
+from api import models
+from api.serializers import UserSerializer
+
+
+class LocationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.Location
+        fields = "__all__"
+
+
+class AircraftSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.Aircraft
+        fields = "__all__"
+
+
+class FlightSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.Flight
+        fields = "__all__"
+        extra_kwargs = {
+            "id": {"read_only": True},
+            "created_at": {"read_only": True},
+            "updated_at": {"read_only": True},
+        }
+
+    def to_representation(self, instance):
+        response = super().to_representation(instance)
+        response["departure"] = LocationSerializer(instance.departure).data
+        response["destination"] = LocationSerializer(instance.destination).data
+        response["aircraft"] = AircraftSerializer(instance.aircraft).data
+
+        return response
+
+    def create(self, validated_data):
+        flight_object = models.Flight.objects.create(**validated_data)
+
+        return flight_object
+
+    def update(self, instance, validated_data):
+        instance.flight_number = validated_data.get(
```

Comment:
any reason to specify each of the values here?

Reply:
yeah, since it is an update method and it allows partial update, so it wont try to save empty fields.

Reply:
You could try this in the viewset to prevent overwriting fields not supplied.
```
def update(self, request, *args, **kwargs):
        kwargs['partial'] = True
        return super(FlightViewSet, self).update(request, *args, **kwargs)

```